### PR TITLE
Add second hand category, Auction Room, Collector shop, entrance-exit, and barrier-turnstile type

### PR DESCRIPTION
**Pull request by endim8**
also remove 'lt' from translate tool, app doesn't build with 'lt' translations

Screenshot_20230605-225818_Organic Maps (Beta)

Screenshot from 2023-06-05 23-01-09
Screenshot from 2023-06-05 23-01-46

### CODE REVIEW

**Comment by endim8** [comment]
fails at https://github.com/organicmaps/organicmaps/actions/runs/5182319500/jobs/9338971679#step:10:51447

> Running categories_test.cpp::CategoriesIndex_AllCategoriesEnglishName
>FAILED
>indexer_tests/categories_test.cpp:329 TEST(index.GetNumTrieNodes() < 10000) 10043 10000 

i'm not really sure what this is, it it enough to just increase the count here?
https://github.com/organicmaps/organicmaps/blob/dec920a06c51d04d6a96285994eca87745d3578b/indexer/indexer_tests/categories_test.cpp#L329

File: android/res/values-de/strings.xml [hunk 1]
```diff
@@ -125,6 +125,8 @@
 	<string name="parking">Parkplatz</string>
 	<!-- Search category for clothes/shoes/gifts/jewellery/sport shops; any changes should be duplicated in categories.txt @shopping! -->
 	<string name="shopping">Shopping</string>
+	<!-- Search category for second_hand/charity/antique/auction shops; any changes should be duplicated in categories.txt @secondhand! -->
+	<string name="secondhand">Aus zweiter hand</string>
```
Comment (Misalf-git) [inline android/res/values-de/strings.xml:129]: Should be "Second-Hand" in german.

File: data/categories.txt [hunk 2]
```diff
@@ -638,6 +638,46 @@ sk:3Banka
...
+# First keyword should match [secondhand] definition in strings.txt!
+@secondhand
+en:6Second Hand
+ar:اليد الثانية
+be:Другая рука
+bg:Втора употреба
+ca:Segona mà
+cs:Z druhé ruky
+da:Brugt
+de:Aus zweiter hand
```
Comment (Misalf-git) [inline data/categories.txt:650]: de:Second-Hand

File: android/res/values-de/strings.xml (hunk 1 above)
Comment (endim8) [reply android/res/values-de/strings.xml]: fixed

File: data/categories.txt [hunk 3]
```diff
@@ -638,6 +638,46 @@ sk:3Banka
...
+ru:Вторая рука
+sk:Z druhej ruky
+sv:Andra handen
+sw:Mtumba
+th:มือสอง
+tr:İkinci el
+uk:Секонд-хенд
+vi:Đồ cũ
+zh-Hans:二手货
+zh-Hant:二手
```
Comment (vng) [inline data/categories.txt:679]: be, ru have strange translation (literally like man's second hand) and I suspect the same on others languages

File: data/categories.txt [hunk 4]
```diff
@@ -11782,7 +11822,7 @@ sk:Obchodník s rybami
 fa:فروشگاه غذای دریایی
 mr:सीफूड शॉप
 
-shop-second_hand|@shopping
+shop-second_hand|@shopping|@secondhand
```
Comment (vng) [inline data/categories.txt:11825]: Better to take translations for secondhand from here.

File: data/strings/strings.txt [hunk 5]
```diff
@@ -2825,6 +2825,47 @@
...
+    ru = Вторая рука
+    sk = Z druhej ruky
+    sv = Andra handen
+    sw = Mtumba
+    th = มือสอง
+    tr = İkinci el
+    uk = Секонд-хенд
+    vi = Đồ cũ
+    zh-Hans = 二手货
+    zh-Hant = 二手
```
Comment (vng) [inline data/strings/strings.txt:2868]: Also be, ru are bad here.

**Comment by vng** [comment]
> > indexer_tests/categories_test.cpp:329 TEST(index.GetNumTrieNodes() < 10000) 10043 10000
> 
> i'm not really sure what this is, is it enough to just increase the count here?

@endim8 This is more or less _a controlling test_ on tracking how our categories index is getting bigger and bigger :). You can increase this constant to 11000.

File: data/categories.txt [hunk 6]
```diff
@@ -15033,3 +15073,83 @@ uk:Електростанція
...
+sv:Auktionsförrättare
+sw:Mnada
+th:ผู้ประมูล
+tr:Müzayedeci
+uk:Аукціоніст
+vi:Đấu giá viên
+zh-Hans:拍卖师
+zh-Hant:拍賣
+
+shop-collector|@shopping|@secondhand
```
Comment (vng) [inline data/categories.txt:15116]: The same here. Is simple "collector" better here?
Category search terms should be short and specific not only for humans but also for the current category search limitations ..

File: data/categories.txt [hunk 7]
```diff
@@ -15033,3 +15073,83 @@ uk:Електростанція
 vi:Nhà máy điện
 zh-Hans:发电厂
 zh-Hant:發電廠
+
+shop-auction|@secondhand
```
[the remark this answers is not in the document]
Comment (endim8) [reply data/categories.txt:15077]: fixed

File: data/categories.txt (hunk 3 above)
Comment (endim8) [reply data/categories.txt:679]: fixed

File: data/categories.txt (hunk 6 above)
Comment (endim8) [reply data/categories.txt:15116]: fixed, but as 'Collectables'

**Comment by endim8** [comment]
hopefully improved translations, increased test limit

**Comment by endim8** [comment]
also added new types for entrance-exit, and barrier-turnstile (this was displayed as a stile before which was an issue)

File: data/categories.txt (hunk 6 above)
Comment (vng) [reply data/categories.txt:15117]: I still prefer _one world_ "collector" here :) Should explain: if we have "Калекцыйныя прадметы" translation in categories, searching for the very generic world "прадметы" will also match this category now.

File: data/categories.txt [hunk 8]
```diff
@@ -15033,3 +15074,84 @@ uk:Електростанція
...
+vi:Bán đấu giá
+zh-Hans:拍卖会
+zh-Hant:拍賣
+
+shop-collector|@shopping|@secondhand
+en:5Collectables
+ar:المقتنيات
+be:Калекцыйныя прадметы
+bg:Колекционерски стоки
+ca:Col·leccionables
```
Comment (vng) [inline data/categories.txt:15122]: Is it a real Catalan translation with a dot?

File: data/strings/types_strings.txt [hunk 9]
```diff
@@ -27084,3 +27082,84 @@
...
+    sv = Auktion
+    sw = Mnada
+    th = ประมูล
+    tr = Müzayede
+    uk = Аукціон
+    vi = Bán đấu giá
+    zh-Hans = 拍卖会
+    zh-Hant = 拍賣
+
+  [type.shop.collector]
```
Comment (vng) [inline data/strings/types_strings.txt:27207]: Here you can keep "Collectables" or any better explanation term unlike in categories.

File: data/mapcss-mapping.csv [hunk 10]
```diff
@@ -747,7 +747,7 @@ amenity|parking_entrance|permissive;[amenity=parking_entrance][access=permissive
 barrier|cattle_grid;747;
 barrier|ditch;748;
 barrier|kissing_gate;749;
-deprecated|deprecated;750;x
+barrier|turnstile;750;
```
Comment (vng) [inline data/mapcss-mapping.csv:750]: Do we really need to distinguish stile and turnstile?

File: data/mapcss-mapping.csv [hunk 11]
```diff
@@ -878,7 +878,7 @@ area:highway|cycleway;877;
 deprecated|deprecated;878;x
 area:highway|motorway;879;
 area:highway|trunk;880;
-deprecated|deprecated;881;x
+entrance|exit;881;
```
Comment (vng) [inline data/mapcss-mapping.csv:881]: Great! Now time to invent some distinguishable (from enter) icon for exits :)

File: data/categories.txt (hunk 8 above)
Comment (endim8) [reply data/categories.txt:15122]: https://www.wikidata.org/wiki/Q208165 (more languages) & https://ca.wikipedia.org/wiki/Col%C2%B7leccionisme
it does seem to be a legitimate translation :)

File: data/mapcss-mapping.csv (hunk 10 above)
Comment (endim8) [reply data/mapcss-mapping.csv:750]: imo it's worth it, stile and turnstile are very different things & I found it quite strange to see stiles at a train station 😄

File: data/categories.txt (hunk 6 above)
Comment (endim8) [reply data/categories.txt:15117]: ah i see what you mean 👍 will fix

**Comment by endim8** [comment]
@vng added this exit icon, will it do? :)
Screenshot from 2023-06-08 14-59-32
also resolved the other review comments

File: data/categories.txt [hunk 12]
```diff
@@ -15033,3 +15074,86 @@ uk:Електростанція
...
+ja:コレクター
+ko:수집가
+mr:कलेक्टर
+nb:Samler
+nl:Verzamelaar
+pl:Kolekcjoner
+pt:Coleccionador
+pt-BR:Coletor
+ro:Colector
+ru:Коллектор
```
Comment (vng) [inline data/categories.txt:15147]: Коллекционер

**Review by vng — COMMENTED** [review]
LGTM.
But! I suspect that we can't merge it now without adding additional "second-hand" into iOS UI.

**Comment by endim8** [comment]
> LGTM. But! I suspect that we can't merge it now without adding additional "second-hand" into iOS UI.

ahh, good point... I don't have a mac any more so adding it will be a challenge, but I'll try 👍

**Comment by endim8** [comment]
fixed ru translation, but i can't add it to iOS unfortunately

**Comment by vng** [comment]
@endim8 Thanks! I will try to update iOS by myself.

**Comment by vng** [comment]
@endim8 Could you please create iOS icons for secondhand, the same way as others (food, transport, ...)?
In next folders:
- iphone/Maps/Images/Search/Categories
- iphone/Maps/Images/Spotlight

**Comment by endim8** [comment]
> @endim8 Could you please create iOS icons for secondhand, the same way as others (food, transport, ...)? 

Done :)

**Comment by vng** [comment]
Great! Please, regenerate symbols (I forgot), and will be ready to merge.

**Comment by endim8** [comment]
Is there an iOS specific symbols generation? Symbols are already regenerated with generate_symbols.sh in https://github.com/organicmaps/organicmaps/pull/5279/commits/1c5c65f17620ae60bb15759827e2149187ced1f5 :+1:

**Pull request merged**